src/v/features/feature_table.cc

```diff
@@ -297,6 +300,30 @@ class feature_table::probe {
     metrics::public_metric_groups _public_metrics;
 };
 
+namespace {
+
+security::license
+make_builtin_trial_license(security::license::clock::time_point start_time) {
+    auto expiry_time = start_time + std::chrono::days{45};
+    auto expiry = std::chrono::duration_cast<std::chrono::seconds>(
+      expiry_time.time_since_epoch());
+
+    if (std::getenv("__REDPANDA_DISABLE_BUILTIN_TRIAL_LICENSE") != nullptr) {
+        // For testing, use an expired trial license
+        expiry = 0s;
+    }
```

Comment:
thought (pedantic): consider two cases 
1. no builtin trial license present (i.e. `nullopt`) 
2. builtin trial license expired

Intuitively these seem distinct, but I'm not sure there's a meaningful difference from a testing perspective. wdyt?

Reply:
I've changed the (so far unused) `revoke_license()` method to also revoke the trial license and added some unit tests around this to demonstrate the behaviour of `should_sanction`. I suspect that `__REDPANDA_DISABLE_BUILTIN_TRIAL_LICENSE` will be enough for ducktape tests (license nag, upgrade blocking tests, etc.), but we can tweak this env var in the future if necessary.